src/App.tsx

```diff
@@ -1,61 +1,127 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { FormEvent, useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+let maxId = [...todosFromServer].sort((a, b) => b.id - a.id)[0].id;
+
+const addTodo = (todo: string, userId: number) => {
+  maxId += 1;
+
+  const todoAdd = {
+    id: maxId,
+    completed: false,
+    title: todo,
+    userId,
+    user: getUser(userId),
+  };
+
+  todos.push(todoAdd);
```

Comment:
Avoid having state outside the component.
We should have `todos` state in the App component